src/vm/util.cpp

```diff
@@ -3154,13 +2856,11 @@ static BOOL TrustMeIAmSafe(void *pLock)
 
 LockOwner g_lockTrustMeIAmThreadSafe = { NULL, TrustMeIAmSafe };
 
-
-DangerousNonHostedSpinLock g_randomLock;
-CLRRandom g_random;
-
-
 int GetRandomInt(int maxVal)
 {
+    static DangerousNonHostedSpinLock s_randomLock;
```

Comment:
Are method-local statics used to be a mine-field because of they are initialized first time the method is called. It means that there are locks that need to be taken by the C/C++ runtime, etc.

I do not know whether they are still a mine-field, but I have been always avoiding them since it is not clear what the code will get compiled into.